src/lib/form-field/form-field.ts

```diff
@@ -266,6 +266,11 @@ export class MatFormField extends _MatFormFieldMixinBase
       this._changeDetectorRef.markForCheck();
     });
 
+    // Run change detection if the prefix or suffix changes.
+    merge(this._prefixChildren.changes, this._suffixChildren.changes).subscribe(() => {
```

Comment:
You should be able to combine these with the `ngControl.valueChanges` subscription from above.